jwql/jwql_monitors/generate_preview_images.py

```diff
@@ -253,7 +258,11 @@ def create_dummy_filename(filelist):
     modules = []
     for filename in filelist:
         indir, infile = os.path.split(filename)
-        det_string = filename_parser(infile)['detector']
+        try:
+            det_string = filename_parser(infile)['detector']
+        except KeyError:
+            # If filename_parser() does not recognize the file, skip it
+            continue
```

Comment:
These lines, on the other hand, imply the filename_parser() will throw an exception for an unrecognized file. Or, possibly, that you still get the dictionary back, but rather than checking for whether it's a recognized filename, you instead just try to access the keyword and continue on an exception.

If feels to me as if, whether or not we want to do the same thing in both cases, we at least want to use the same idiom in calling filename_parser(). Here, that might look like

```
file_parts = filename_parser(infile)
if not file_parts['recognized_filename']:
    continue
det_string = file_parts['detector']
```

Reply:
My second note is that, on failure, we absolutely need to log an error (or warning), so that we know that there's something that's causing the failures. We don't want to crash, but we *do* want to know.